### PR TITLE
CASMTRIAGE-8101 - get mountain ssh keys locally.

### PR DESCRIPTION
**Pull request by dlaine-hpe**
## Summary and Scope

The ssh keys to access the mountain consoles had been stored on a shared pvc volume, but something would change the permissions of the file every once in a while. If the permissions changed ssh would no longer use the key until the permissions were restored to `600`.

This change removes the keys from the shared volume and each container now retrieves the keys directly from vault and stores them on local volumes. This should resolve the changing permissions issue for the key.

I also updated dependency versions to resolve security issues.

## Issues and Related PRs
* Resolves [CASMTRIAGE-8101](https://jira-pro.it.hpe.com:8443/browse/CASMTRIAGE-8101)

### Tested on:
  * `Rocket`

### Test description:

The new versions of console-operator and console-node were installed via helm, tested to insure all is functioning correctly, then reverted to the previous version.

- Were the install/upgrade-based validation checks/tests run (goss tests/install-validation doc)? N
- Were continuous integration tests run? If not, why? N
- Was upgrade tested? If not, why? Y
- Was downgrade tested? If not, why? Y
- Were new tests (or test issues/Jiras) created for this change? N

## Risks and Mitigations

Fairly low risk as this is just fixing some edge cases.

## Pull Request Checklist
- [X] Version number(s) incremented, if applicable
- [X] Copyrights updated
- [X] License file intact
- [X] Target branch correct
- [X] CHANGELOG.md updated
- [X] Testing is appropriate and complete, if applicable
- [X] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

File: src/console_node/creds.go [hunk 1]
```diff
@@ -109,41 +115,195 @@ func getPasswords(bmcXNames []string) map[string]compcreds.CompCredentials {
 	return ccreds
 }
 
+// Location of the Kubernetes service account token used to authenticate
+// to Vault.  This is part of the pod deployment.
+const svcAcctTokenFile string = "/var/run/secrets/kubernetes.io/serviceaccount/token"
+
+// The Vault base URI
+const vaultBase = "http://cray-vault.vault:8200/v1"
```
Comment (kumarrahul04) [inline src/console_node/creds.go:123]: would be better if we put this in deployment YAML file.

File: src/console_node/creds.go [hunk 2]
```diff
@@ -109,41 +115,195 @@ func getPasswords(bmcXNames []string) map[string]compcreds.CompCredentials {
...
+			"Unable to get the %s secret from vault: %s  Error was: %s",
+			vaultBmcKeyName, vaultBase, err)
+		return "", response, responseCode, fmt.Errorf("Unable to get the %s secret from vault: %s  Error was: %s",
+			vaultBmcKeyName, vaultBase, err)
+	}
+
+	if responseCode == 404 {
+		log.Printf("The vault secret %s was not found. It will need to be created.", vaultBmcKeyName)
+
+		return "", response, 404, nil
```
Comment (kumarrahul04) [inline src/console_node/creds.go:151]: better to use http package constants for http return codes.

File: src/console_node/creds.go [hunk 3]
```diff
@@ -109,41 +115,195 @@ func getPasswords(bmcXNames []string) map[string]compcreds.CompCredentials {
...
+
+		// If the response code is not 200 then we failed authentication.
+		if responseCode == 200 {
+			log.Printf("Vault authentication was successful.  Attempting to get BMC console key from vault")
+			vaultToken = gjson.Get(string(response), "auth.client_token").String()
+			break
+		}
+
+		// Can't access vault, wait and try again
+		log.Printf("Unable to authenticate with vault, will try again in 15 seconds")
```
Comment (kumarrahul04) [inline src/console_node/creds.go:233]: is it to haldle if valut is not up?

File: src/console_node/creds.go [hunk 4]
```diff
@@ -109,41 +115,195 @@ func getPasswords(bmcXNames []string) map[string]compcreds.CompCredentials {
...
 		// keys do not exist, wait and try again
 		log.Printf("Mountain ssh keys do not exist - waiting for them...")
 		time.Sleep(5 * time.Second)
 	}
 }
+
+// Ensure that Mountain node console key files are present.
+//func ensureMountainConsoleKeysPresent() {
+// NOTE: in update config thread
+
```
Comment (kumarrahul04) [inline src/console_node/creds.go:275]: these commented lines are needed?

Comment (dlaine-hpe) [reply src/console_node/creds.go:275]: Ooops - forgot to remove the old version. Good catch - removed.

File: src/console_node/creds.go (hunk 3 above)
Comment (dlaine-hpe) [reply src/console_node/creds.go:233]: Traditionally there have been enough network issues that a retry is a good idea for critical stuff. Also the pods may be getting restarted or relocated, or other things that k8s can do to temporarily interrupt a service.

File: src/console_node/creds.go [hunk 5]
```diff
@@ -109,41 +115,195 @@ func getPasswords(bmcXNames []string) map[string]compcreds.CompCredentials {
...
-
-		if !noKey {
-			// keys exist so done
-			log.Printf("Mountain ssh key files exist")
-			return
+		// Authenticate to Vault using the service account token
+		vaultAuthParam := map[string]string{
+			"jwt":  string(svcAcctToken),
+			"role": "ssh-user-certs-compute"}
+		jsonVaultAuthParam, _ := json.Marshal(vaultAuthParam)
```
[the remark this answers is not in the document]
Comment (dlaine-hpe) [reply src/console_node/creds.go:220]: Handled.

File: src/console_node/creds.go (hunk 1 above)
Comment (dlaine-hpe) [reply src/console_node/creds.go:123]: Not sure if it will ever be used, but possible so done.
